cocos/base/CCAsyncTaskPool.h

```diff
@@ -89,11 +89,19 @@ class CC_DLL AsyncTaskPool
      * @param type task type is io task, network task or others, each type of task has a thread to deal with it.
      * @param callback callback when the task is finished. The callback is called in the main thread instead of task thread.
      * @param callbackParam parameter used by the callback.
-     * @param f task can be lambda function.
+     * @param task: task can be lambda function to be performed off thread.
      * @lua NA
      */
-    template<class F>
-    inline void enqueue(TaskType type, const TaskCallBack& callback, void* callbackParam, F&& f);
+    void enqueue(TaskType type, TaskCallBack callback, void* callbackParam, std::function<void()> task);
```

Comment:
We should remove the template<class F>.

If we make if an l-value reference, then doing something like

```
AsyncTaskPool::getInstance()->enqueue(0, myFunction, myParams, [] () {

});
```

Won't compile, since the lambda will be a r-value. And we really don't get anything making it an l-value reference over taking it by value.